### PR TITLE
Add billing agreement details to PayPalCheckoutRequest

### LINKED ISSUE #509: Custom PayPalCheckoutRequest billing agreement description text not displaying (api:paypal:4.8.1)
### General information

* SDK/Library version:

4.8.1

Screen Shot 2022-01-31 at 12 33 01 PM

* Environment:

Sandbox

* Android Version and Device:

Physical OnePlus 8 Pro running Android 11

My app's SDK versions 👇🏽 

Screen Shot 2022-01-31 at 12 52 19 PM

* Braintree dependencies:

`implementation com.braintreepayments.api:paypal:4.+`

`implementation com.braintreepayments.api:data-collector:4.+`

### Issue description

It seems that a custom billing description string isn't being displayed at the top of the payment selection web page during an Android Checkout payment flow. I've set up my flow according to https://developer.paypal.com/braintree/docs/guides/paypal/checkout-with-paypal 

I've tried going through the flow with a personal *sandbox* account **and** a business *sandbox* account. The text doesn't display in neither situation.

I'm trying to get the `Preferred Funding...` sentence to show below the `Hi, John` area.

**On Android:** 👇🏽

Screen Shot 2022-01-31 at 12 47 58 PM

The same sentence is displayed for my iOS colleague when he set up the same flow on iOS 👇🏽

Screen Shot 2022-01-31 at 12 35 43 PM

This is how I'm building my `PayPalCheckoutRequest` object.
Screen Shot 2022-01-31 at 12 24 10 PM

I logged the description string (`getStringOrEmpty(R.string.booking_payment_fragment_paypal_billing_description` seen in the screenshot above) to make sure that I wasn't erroneously using a blank string when building the `PayPalCheckoutRequest` object. It appears to be fine in the logcat 👇🏽

Screen Shot 2022-01-28 at 3 44 24 PM (1)

I also checked the request's billing description before I actually tokenize the PayPal account 👇🏽

Screen Shot 2022-01-31 at 12 45 14 PM

Screen Shot 2022-01-31 at 12 44 54 PM

Any of you know what might be going wrong? A known bug?

### PR DESCRIPTION
**Pull request by jaxdesmarais**
### Summary of changes

 - Add billing agreement details to PayPalCheckoutRequest
 - Update unit tests
 - Fixes #509

#### Testing
Because the sandbox account in the demo app uses the mocked payment flow, we are not able to add an integration test for this. We were able to manually verify by doing the following:

Added the following in the demo app:
```
request.setShouldRequestBillingAgreement(true);
request.setBillingAgreementDescription("A random description");
```

The description now shows as expected:

 ### Checklist

 - [x] Added a changelog entry

- @sarahkoop @jaxdesmarais

### CODE REVIEW

File: PayPal/src/main/java/com/braintreepayments/api/PayPalCheckoutRequest.java [hunk 1]
```diff
@@ -177,7 +177,9 @@ String createRequestBody(Configuration configuration, Authorization authorizatio
 
         String billingAgreementDescription = getBillingAgreementDescription();
         if (shouldRequestBillingAgreement && !TextUtils.isEmpty(billingAgreementDescription)) {
-            parameters.put(DESCRIPTION_KEY, billingAgreementDescription);
```
Comment (sshropshire) [inline PayPal/src/main/java/com/braintreepayments/api/PayPalCheckoutRequest.java:180]: random q: was this the wrong key? I wish I could :actual-lol: here lol

Comment (sarahkoop) [reply PayPal/src/main/java/com/braintreepayments/api/PayPalCheckoutRequest.java:180]: The key was write - we were just missing a layer of nesting in the JSON

**Comment by langsmith** [comment]
As the author of https://github.com/braintree/braintree_android/issues/509, I'm glad that this is getting resolved! Thanks everyone 😃

**Review by jcnoriega — APPROVED** [review]

**Comment by langsmith** [comment]
Do ya'll know when the next SDK release, which includes this pr fix, will go out?

**Comment by langsmith** [comment]
> Do ya'll know when the next SDK release, which includes this pr fix, will go out?

Thoughts  @sshropshire @sarahkoop  ?

**Comment by sarahkoop** [comment]
@langsmith this change has been released in `4.8.3`.

**Comment by langsmith** [comment]
Great, thanks!

**Pull request merged**